src/lib/isTaxID.js

```diff
@@ -1136,6 +1136,10 @@ const taxIdFormat = {
   'en-CA': /^\d{9}$/,
   'en-GB': /^\d{10}$|^(?!GB|NK|TN|ZZ)(?![DFIQUV])[A-Z](?![DFIQUVO])[A-Z]\d{6}[ABCD ]$/i,
   'en-IE': /^\d{7}[A-W][A-IW]{0,1}$/i,
+  'en-IN': {
+    GSTIN: /^((?!39)(?!00)[0-3][0-9]|97)([A-Z]{3}[ABCFGHLJPT][A-Z](?!0000)[0-9]{4}[A-Z])[1-9A-Z]Z[0-9 A-Z]$/,
```

Comment:
Do you have an additional source for the GSTIN build up? You did link to the state code and the PAN number, but not to the entire number

Reply:
You could refer the below link
[Structure of GSTIN from Indirect Tax Committee of India  ](http://idtc-icai.s3.amazonaws.com/download/knowledgeShare18-19/Structure-of-GSTIN.pdf)
Please note I have made gstin validator only for Regular Registration GSTIN which is used widely  and not for NRTP type GSTIN, might add it in future.